File: src/test/java/com/testainers/MethodsResourceTest.java
package com.testainers;

import io.quarkus.test.junit.QuarkusTest;
import io.restassured.http.Method;
import org.junit.jupiter.api.Test;

import static org.hamcrest.Matchers.*;

/**
 * @author Eduardo Folly
 */
@QuarkusTest
public class MethodsResourceTest extends BaseResourceTest {

    @Test
    public void methodsGet() {
        base()
                .get("/methods")
                .then()
                .statusCode(200)
                .body("", not(""), bodyMatchers(Method.GET));
    }

    @Test
    public void methodsHead() {
        base().head("/methods")
              .then()
              .statusCode(200);
    }

    @Test
    public void methodsPost() {
        json(Method.POST)
                .post("/methods")
                .then()
                .statusCode(200)
                .body("body", is(BODY), bodyMatchers(Method.POST));
    }

    @Test
    public void methodsPut() {
        json(Method.PUT)
                .put("/methods")
                .then()
                .statusCode(200)
                .body("body", is(BODY), bodyMatchers(Method.PUT));
    }

    @Test
    public void methodsPatch() {
        json(Method.PATCH)
                .patch("/methods")
                .then()
                .statusCode(200)
                .body("body", is(BODY), bodyMatchers(Method.PATCH));
    }

    @Test
    public void methodsDelete() {
        json(Method.DELETE)
                .delete("/methods")
                .then()
                .statusCode(200)
                .body("body", is(BODY), bodyMatchers(Method.DELETE));
    }

}

File: src/test/kotlin/com/testainers/MethodsResourceTest.kt
package com.testainers

import io.quarkus.test.junit.QuarkusTest
import io.restassured.http.Method
import org.hamcrest.Matchers.*
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.MethodSource

/**
 * @author Eduardo Folly
 */
@QuarkusTest
class MethodsResourceTest : BaseResourceTest() {
    @ParameterizedTest
    @MethodSource("onlyMethods")
    fun methodGet(method: Method) {
        json(method)
            .request(method, "/methods")
            .then()
            .statusCode(200)
            .apply {
                if (method != Method.HEAD) {
                    body(
                        "body",
                        if (method == Method.GET) {
                            nullValue()
                        } else {
                            equalTo(body)
                        },
                        *bodyMatchers(method),
                    )
                }
            }
    }
}
